### PR TITLE
Add tclLsp.diagnostics.exclude glob patterns for file-level suppression

### PR DESCRIPTION
**Pull request by bitwisecook**
## Summary

Implements issue #1556: adds `tclLsp.diagnostics.exclude` configuration to suppress all diagnostics for files matching glob patterns, without requiring per-code suppression directives.

- **New module `path_glob`**: Hand-rolled gitignore-style glob matcher supporting `*`, `?`, `[...]` character classes, `**` globstars, brace alternation `{a,b}`, and escapes. Matches case-sensitively with `/` as an unmatchable separator.
- **Configuration integration**: `[diagnostics] exclude` in `.tcl-lsp.ini` or `config.ini` accepts one glob pattern per line. Follows standard precedence: project config overrides editor settings overrides global config. Per-folder in multi-root workspaces.
- **Diagnostic pipeline**: Excluded files publish empty diagnostic sets (clearing existing squiggles) via a new `run_diagnostics_excluded` path, with a `[timing] diagnostics excluded` marker for reliable test barriers.
- **Live reload**: Changes to `.tcl-lsp.ini` or editor settings take effect immediately without restart.

Patterns with `/` match workspace-folder-relative paths; patterns without `/` match file names at any depth (gitignore-style). A trailing `/` is directory sugar (`vendor/` → `vendor/**`). Malformed patterns (unterminated `[` class, brace expansion exceeding 1024 alternatives) are silently dropped.

## Validation

- Added comprehensive unit tests in `path_glob.rs` covering name/path patterns, globstars, character classes, brace alternation, escapes, and edge cases.
- Added e2e tests (`issue1556_diagnostics_exclude.rs`) verifying editor config exclusion and project `.tcl-lsp.ini` live reload.
- Added KCS documentation (`kcs-howto-exclude-files-from-diagnostics.md`) with pattern syntax reference and configuration examples.
- Updated README and design contracts to document the feature.

## Compiler/KCS checklist

- [x] Added new KCS how-to guide: `docs/kcs/kcs-howto-exclude-files-from-diagnostics.md`
- [x] Linked from `docs/kcs/README.md`
- [x] Updated design contract: `docs/design/contracts/lsp-diagnostics-publication.md`
- [x] Updated README with configuration example

https://claude.ai/code/session_01GY7KNgPPz5WPrC6WzsHMqT

### CODE REVIEW

File: docs/kcs/kcs-howto-exclude-files-from-diagnostics.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# KCS: How do I turn off all diagnostics for certain files?
+
+> **Audience:** User
+> **Type:** How-To
```
[the remark this answers is not in the document]
Comment (bitwisecook) [reply docs/kcs/kcs-howto-exclude-files-from-diagnostics.md:4]: Fixed in 9aeb6c17: `kcs-feature-diagnostics.md` now documents file-level exclusion in the Settings bullet and Operational context, and links `kcs-howto-exclude-files-from-diagnostics.md` from its Discoverability section.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: rust/tcl-lsp-server/src/lib.rs [hunk 2]
```diff
@@ -11806,6 +11849,27 @@ impl Backend {
...
+        // `tclLsp.diagnostics.exclude` — glob patterns for files that produce
+        // no diagnostics at all (#1556). Reset unconditionally: the production
+        // caller always passes the fully merged config, so an absent key means
+        // the exclusion list really is empty now — a stale list must not keep
+        // suppressing a file the user just un-excluded.
+        let exclude: Vec<String> = cfg
+            .get("diagnostics")
+            .and_then(serde_json::Value::as_object)
+            .and_then(|d| d.get("exclude"))
+            .and_then(serde_json::Value::as_array)
```
[the remark this answers is not in the document]
Comment (bitwisecook) [reply rust/tcl-lsp-server/src/lib.rs:11861]: Fixed in 9aeb6c17: `tclLsp.diagnostics.exclude` is now generated into VS Code's `package.json` (via `gen_vscode_package.rs`, under *Diagnostics — Errors*), with the workspace/configuration middleware dropping the empty-array default so an unconfigured folder can't replace an ini-layer list — same absent-vs-empty treatment as `genericVariablePatterns`. JetBrains gained a `diagnosticsExclude` settings field (panel row + `toServerSettings()` serialization, omitted when blank for the same reason). Kotlin and TS both compile; the package.json drift gate passes.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: rust/tcl-lsp-server/src/lib.rs [hunk 3]
```diff
@@ -2996,6 +3022,9 @@ async fn run_diagnostics_core(inputs: DiagInputs, uri: &Uri, job: DiagJob) -> bo
     if !toggles.diagnostics_enabled {
         return run_diagnostics_master_off(&delivery).await;
     }
+    if toggles.excluded {
+        return run_diagnostics_excluded(&delivery).await;
+    }
```
[the remark this answers is not in the document]
Comment (bitwisecook) [reply rust/tcl-lsp-server/src/lib.rs:3027]: Fixed in 9aeb6c17: `publish_spec_pack_notices` now drops excluded URIs from the batch before the stale-set computation, so exclusion both suppresses new pack notices and clears already-published ones on the next reload — and adding an exclusion is a config apply, which triggers that reload (`ReloadTrigger::Config`). Covered by a new e2e test (`diagnostics_exclude_suppresses_pack_load_notices`) that sees the `SPECTCL` badge appear and then clear when `*.tclspec` is excluded; the contract doc records the gate.

---
_Generated by [Claude Code](https://claude.ai/code)_
